sematic/resolvers/log_streamer.py

```diff
@@ -251,22 +261,30 @@ def ingested_logs(
 
     original_signal_handler = None
     streamer_pid = None
+    read_file_descriptor = None
+    write_file_descriptor = None
 
-    def clean_up_streamer(signal_num, frame=None):
+    def clean_up_streamer(signal_num=None, frame=None):
         logger.info("Cleaning up log ingestor")
+        print(_TERMINATION_CHAR)  # tell the reader that the stream is done
+
+        # ensure there's a final log upload, and that it contains ALL the
+        # contents of stdout and stderr before we redirect them back to their
+        # originals.
+        sys.stdout.flush()
+        sys.stderr.flush()
```

Comment:
Please flush stderr first. Both because stderr is in principle considered higher priority, and because the EOT is written to stdout, and theoretically there can be a race condition where the last buffer's worth of stderr is missed once the EOT from stdout is processed.